protos/sigstore_trustroot.proto

```diff
@@ -88,6 +89,57 @@ message TransparencyLogInstance {
         // e.g. two proofs from log deployments operated by the same operator
         // should count as only one valid proof.
         string operator = 6;
+        // The list of witnesses the log will be witnessed by along with the
+        // quorum policy.
+        // Only supported for TrustedRoot media types matching or greater than
+        // application/vnd.dev.sigstore.trustedroot.v0.2+json
+        WitnessConfiguration witness_config = 7;
+        // When set, signifies to a client that the log is frozen and no longer
+        // is accepting entries. When verifying an inclusion proof, the client
+        // SHOULD use this checkpoint and verify that the log index in the bundle
+        // is not greater than or equal to the checkpoint tree size.
+        // Only supported for TrustedRoot media types matching or greater than
+        // application/vnd.dev.sigstore.trustedroot.v0.2+json
+        dev.sigstore.rekor.v1.Checkpoint frozen_log_checkpoint = 8;
+        // Additional public keys used to verify log checkpoint signatures for when
+        // a log uses multiple signing algorithms to generate checkpoint signatures.
+        // Only supported for TrustedRoot media types matching or greater than
+        // application/vnd.dev.sigstore.trustedroot.v0.2+json
+        repeated dev.sigstore.common.v1.PublicKey additional_public_keys = 9;
+}
+
+// WitnessConfiguration contains the list of witnesses that will verify the
+// consistency of the log. Witnesses are grouped together, with each group
+// specifying its own quorum rule, and the list of groups needing to meet a
+// quorum rule. Having these groupings allows for more complex witnessing
+// policies, e.g. requiring 2-of-3 from 1) a set of ArmoredWitness instances
+// requiring M-of-N, 2) any from a set of regionalized witnesses run by a
+// single operator, and 3) a single witness instance.
+// Inspired by https://git.glasklar.is/sigsum/core/sigsum-go/-/blob/main/doc/policy.md
+message WitnessConfiguration {
+        // The list of witness groups the log will be witnessed by.
+       repeated WitnessGroup witness_groups = 1 [(google.api.field_behavior) = REQUIRED]; 
+        // Specifies how many witness groups are required for proving consistency.
+        ServiceConfiguration service_config = 2 [(google.api.field_behavior) = REQUIRED];
```

Comment:
It might also be worthwhile to allow specifying a witness group name or witness name (or list of such) as they do in the sigsum policy.